### PR TITLE
[Flang][OpenMP] Disable all OpenMP semantics tests on Windows

### PR DESCRIPTION
**Pull request by kiranchandramohan**
Removes two XFAILed tests, the other tests are marked UNSUPPORTED only on windows.

### CODE REVIEW

File: flang/test/Semantics/OpenMP/do02.f90 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ! RUN: %S/test_errors.sh %s %t %flang -fopenmp
-! XFAIL: *
+! UNSUPPORTED: *
```
Comment (kiranchandramohan) [inline flang/test/Semantics/OpenMP/do02.f90:2]: * does not work with UNSUPPORTED.

**Comment by Endilll** [comment]
To be clear where this specific list of tests came from:
1. `do02.f90` is for sure know to cause the CI pause (https://discourse.llvm.org/t/flang-tests-are-extremely-slow-on-windows/78591/11)
2. `taskloop01.f90`, `taskloop02.f90`, `taskloop03.f90`, and `taskwait.f90` is a small set of tests I ran out of time to bisect, so they are not individually proven to be problematic.

**Comment by Endilll** [comment]
It's also worth noting that the list is not exhaustive. I'd be rather surprised if this would fix our Windows CI problems, but it's a step in the right direction.

**Review by tblah — COMMENTED** [review]
Please could you add a comment with the UNSUPPORTED directive to explain that these tests are intended to work but they may be one of the tests slowing down CI.

**Comment by kiranchandramohan** [comment]
> It's also worth noting that the list is not exhaustive. I'd be rather surprised if this would fix our Windows CI problems, but it's a step in the right direction.

Yes. This alone did not help. I will try disabling all Semantics/OpenMP tests to see whether it helps. If it helps then I will reduce the disabled ones using the SemanticsChecked list.

**Review by tblah — APPROVED** [review]
This LGTM so long as it does fix windows CI

**Comment by Endilll** [comment]
> > It's also worth noting that the list is not exhaustive. I'd be rather surprised if this would fix our Windows CI problems, but it's a step in the right direction.
> 
> Yes. This alone did not help. I will try disabling all Semantics/OpenMP tests to see whether it helps. If it helps then I will reduce the disabled ones using the SemanticsChecked list.

Yes, that should resolve the issue with CI. I tested that yesterday.

**Comment by kiranchandramohan** [comment]
> > > It's also worth noting that the list is not exhaustive. I'd be rather surprised if this would fix our Windows CI problems, but it's a step in the right direction.
> > 
> > 
> > Yes. This alone did not help. I will try disabling all Semantics/OpenMP tests to see whether it helps. If it helps then I will reduce the disabled ones using the SemanticsChecked list.
> 
> Yes, that should resolve the issue with CI. I tested that yesterday.

This (disabling all Semantics OpenMP tests) worked. So I am going to commit this now. Will reenable the tests based on the SemanticsChecked tag in a separate PR.

**Pull request merged**